vsts.yml

```diff
@@ -26,71 +21,81 @@ phases:
       # better solution for checking out the commit to be built.
       (cd src/electron; git fetch origin +"${BUILD_SOURCEBRANCH}"; git checkout "${BUILD_SOURCEVERSION}")
       gclient sync --with_branch_heads --with_tags
-    name: gclient_sync
+      cd src
+      export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
+      echo "##vso[task.setvariable variable=CHROMIUM_BUILDTOOLS_PATH]`pwd`/buildtools"
+    displayName: 'gclient sync'
 
   - bash: |
       cd src
-      export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
-      export SCCACHE_BINARY="`pwd`/electron/external_binaries/sccache"
-      # SCCACHE_AZURE_BLOB_CONTAINER and SCCACHE_AZURE_CONNECTION_STRING are expected to be set.
-      "$SCCACHE_BINARY" --start-server
-      echo "##vso[task.setvariable variable=SCCACHE_BINARY]$SCCACHE_BINARY"
-      echo "##vso[task.setvariable variable=CHROMIUM_BUILDTOOLS_PATH]$CHROMIUM_BUILDTOOLS_PATH"
+      export SCCACHE_HELPER="`pwd`/electron/script/sccache-helper.py"
+      "$SCCACHE_HELPER" --start-server --aws-access-key "$(SCCACHE_AWS_ACCESS_KEY)" --aws-secret "$(SCCACHE_AWS_SECRET)"
+      export SCCACHE_WRAPPER="`pwd`/electron/external_binaries/sccache"
+      echo "##vso[task.setvariable variable=SCCACHE_WRAPPER]$SCCACHE_WRAPPER"
+      "$SCCACHE_WRAPPER" -s
       echo "GN gen for: $GN_CONFIG"
-      gn gen out/Default --args='import("'$GN_CONFIG'") cc_wrapper="'"$SCCACHE_BINARY"'"'
-    name: GN_gen
+      gn gen out/Default --args='import("'$GN_CONFIG'") cc_wrapper="'"$SCCACHE_WRAPPER"'"'" $GN_EXTRA_ARGS"
+    displayName: 'GN gen with sccache'
+    condition: and(succeeded(), ne(variables['ELECTRON_RELEASE'], '1'))
+
+  - bash: |
+      cd src
+      echo "GN gen for: $GN_CONFIG"
+      gn gen out/Default --args='import("'$GN_CONFIG'")'" $GN_EXTRA_ARGS"
+    displayName: 'GN gen without sccache'
+    condition: and(succeeded(), eq(variables['ELECTRON_RELEASE'], '1'))
 
   - bash: |
       cd src
       ninja -C out/Default electron:electron_app
-    name: Ninja_build
+    displayName: 'Ninja build app'
 
   - bash: |
-      "$SCCACHE_BINARY" -s
-      "$SCCACHE_BINARY" --stop-server
-    name: Check_sccache_stats
+      "$SCCACHE_WRAPPER" -s
+      "$SCCACHE_WRAPPER" --stop-server
+    displayName: 'Check sccache stats after build'
+    condition: and(succeeded(), ne(variables['ELECTRON_RELEASE'], '1'))
 
   - bash: |
       set +e
       cd src
       ninja -C out/Default third_party/electron_node:headers
       export npm_config_nodedir="$PWD/out/Default/gen/node_headers"
       (cd electron/spec && npm install)
-      ./out/Default/Electron.app/Contents/MacOS/Electron electron/spec --ci --enable-logging
-    name: Test
-    condition: and(succeeded(), ne(variables['ELECTRON_RELEASE'], '1'))
+      ./out/Default/Electron.app/Contents/MacOS/Electron electron/spec --ci
```

Comment:
why remove `--enable-logging`?

Reply:
because --enable-logging causes a dcheck to fail on one of the sandbox tests.

Reply:
Its this one https://github.com/electron/electron/issues/13770#issuecomment-412199748 , chrome provides an absolute path for the log file https://cs.chromium.org/chromium/src/chrome/common/logging_chrome.cc?sq=package:chromium&g=0&l=274-277 , in this scenario. We don't https://github.com/electron/electron/blob/master/atom/app/atom_main_delegate.cc#L77

Reply:
@deepak1556  yes that one.

Reply:
yup only windows